### ISSUE TITLE
Derivation of variance

### ISSUE THREAD

**Issue by mkutny** [pgte/moving-average#6]
Hi,

just curious how did you get to your version of variance calculation:
`v + (value - previousMa) * (value  - ma);`

From what I read [around](http://people.ds.cam.ac.uk/fanf2/hermes/doc/antiforgery/stats.pdf) it should be `a * v + (1 - a) * а * squared(value - previousMa)`.

I see there is a `squared` function defined in your code which you don't use - I guess it was designed for that reason?

**Comment by pgte** [comment]
I've read some different variations of implementing moving variance.
I'm ok with changing it to a more consensual one.
Is there a consensus on which one is better?

**Comment by mkutny** [comment]
Well, I truly believe that even if equations are written differently they should give the same results.

I tried to derive your version given that one I've found but it seems to me they are different.

It might be that I made a mistake in derivation or it might be that your formula is wrong. Therefore I was interested to look at the source of it.

The previously quoted paper and in [Wiki](https://en.wikipedia.org/wiki/Moving_average#Exponentially_weighted_moving_variance_and_standard_deviation) they both provide the following online algorithm:
```
diff := x - mean
incr := alpha * diff
mean := mean + incr
variance := (1 - alpha) * (variance + diff * incr)
```
Mean calculation here is equivalent to yours version of it but I can't derive your version of variance from the given above.

**Comment by pgte** [comment]
@mkutny yeah, couldn't derive this neither. I should have documented how I derived this from.
Anyway, could you take a look at a6fa08d3693ddbbcf01bbc38bc30a58417f2dde7 and see if this looks ok?

**Comment by mkutny** [comment]
Looking at diff just noticed that previously upon returning variance it was divided by the number of samples. I totally missed that!

Calculations look good now, thanks!

**Comment by pgte** [comment]
@mkutny cool, thanks for the feedback!

Landed on v0.2.0.
